packages/rancher-windows-gmsa/charts/templates/credentialspec.yaml

```diff
@@ -0,0 +1,25 @@
+{{- if .Values.credential.enabled -}}
+apiVersion: windows.k8s.io/v1
+kind: GMSACredentialSpec
+metadata:
+  name: {{ .Values.credential.domainJoinConfig.machineAccountName | lower }}

```

Comment:
what if the machineAccountName is not provided? Since these are all required values, we should use the [`required`](https://austindewey.com/2018/12/28/helm-tricks-input-validation-with-required-and-fail/) functionality in Helm to ensure this fails on running `helm template` if these values are not provided.

This can be done within some files (like here) or done as a call within a dummy file, just like we do for `validate-install-crd` here: https://github.com/rancher/charts/blob/73f1cd647751650190644b23d0e41221bdfed644/charts/rancher-monitoring/templates/validate-install-crd.yaml#L16

Reply:
@aiyengar2 are we able to make something required that could be disabled?

Reply:
yes, for example in the link above, we have an if statement gating the required block:

https://github.com/rancher/charts/blob/73f1cd647751650190644b23d0e41221bdfed644/charts/rancher-monitoring/templates/validate-install-crd.yaml#L1

This is what ensures that a `helm template` (which finds 0 cluster roles since it's not running in a real cluster) does not fail on not finding the CRDs installed.

Reply:
I don't think it should be required. If they are enabling it they would set the values. If the values aren't set, then the Credential just wouldn't work, but wouldn't prevent the whole from being available.